### PR TITLE
[Merged by Bors] - feat: port Algebra.Tropical.BigOperators

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Algebra/Tropical/BigOperators.lean [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+collection of linear functions.
+
+## Main declarations
+
+* `untrop_sum`
+
+## Implementation notes
+
+No concrete (semi)ring is used here, only ones with inferrable order/lattice structure, to support
+real, rat, ereal, and others (erat is not yet defined).
```
Comment (jcommelin) [inline Mathlib/Algebra/Tropical/BigOperators.lean:32]:
```suggestion
`Real`, `Rat`, `EReal`, and others (`ERat` is not yet defined).
```

File: Mathlib/Algebra/Tropical/BigOperators.lean [hunk 2]
```diff
@@ -114,35 +115,36 @@ theorem trop_infᵢ [ConditionallyCompleteLinearOrder R] [Fintype S] (f : S →
...
+    rfl
 #align multiset.untrop_sum Multiset.untrop_sum
 
 theorem Finset.untrop_sum' [LinearOrder R] [OrderTop R] (s : Finset S) (f : S → Tropical R) :
     untrop (∑ i in s, f i) = s.inf (untrop ∘ f) := by
-  cases s
-  convert Multiset.untrop_sum _
-  simpa
+  convert Multiset.untrop_sum (s.val.map f)
+  simp
```
Comment (jcommelin) [inline Mathlib/Algebra/Tropical/BigOperators.lean:128]: Non-terminal simp, please replace it with the output of `simp?`.

Comment (xroblot) [reply Mathlib/Algebra/Tropical/BigOperators.lean:128]: Oh, that's an error. I didn't push the latest version...

BTW, is something like that okay to finish a proof?  
```
simp 
rfl
```

Comment (jcommelin) [reply Mathlib/Algebra/Tropical/BigOperators.lean:128]: It's probably harmless, but I would still prefer the `simp only` version.

**Comment by jcommelin** [comment]
bors d+

**Pull request closed without merging**